src/components/button/button.tsx

```diff
@@ -0,0 +1,76 @@
+interface ButtonCarreteProps {
+  type: string;
```

Comment:
el type es un "string" puede ser cualquier string, entonces is esta prop tomara un valor diferente al de los definidos por los cases del switch, va a parar al caso del "default"